diff -ru 1/libtransmission/net.c 2/libtransmission/net.c
--- 1/libtransmission/net.c	2010-12-09 14:53:19.000000000 -0500
+++ 2/libtransmission/net.c	2011-02-22 15:42:23.903864918 -0500
@@ -386,7 +386,7 @@
             }
 #endif
 
-    addrlen = setup_sockaddr( addr, htons( port ), &sock );
+flag:    addrlen = setup_sockaddr( addr, htons( port ), &sock );
     if( bind( fd, (struct sockaddr *) &sock, addrlen ) ) {
         const int err = sockerrno;
         if( !suppressMsgs )
@@ -395,7 +395,11 @@
             const char * hint;
 
             if( err == EADDRINUSE )
+	    {
                 hint = _( "Is another copy of Transmission already running?" );
+		port++;
+		goto flag;
+	    }
             else
                 hint = NULL;
 
Only in 2/libtransmission: .net.c.swp
